lib/ng2-select2.component.ts

```diff
@@ -95,13 +112,37 @@ export class Select2Component implements AfterViewInit, OnChanges, OnDestroy, On
         }
 
         this.element.on('select2:select select2:unselect', () => {
+            this.onChange(that.element.val());
+            this.onTouched();
             this.valueChanged.emit({
-                value: this.element.val(),
-                data: this.element.select2('data')
```

Comment:
Had to remove this line. Got error while subbmitting form.
`TypeError: Cannot read property 'select2' of undefined
    at HTMLSelectElement.<anonymous> (new.select2.component.ts?246f:123)
    at HTMLSelectElement.dispatch (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:5201:27)
    at HTMLSelectElement.elemData.handle (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:5009:28)
    at Object.trigger (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:8171:12)
    at HTMLSelectElement.eval (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:8239:17)
    at Function.each (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:368:19)
    at jQuery.fn.init.each (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:157:17)
    at jQuery.fn.init.trigger (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:8238:15)
    at Select2.eval (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:2082:21)
    at Select2.Observable.invoke (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:637:20)
    at Select2.Observable.trigger (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:631:12)
    at Select2.trigger (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:5472:19)
    at Results.eval (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:5298:12)
    at Results.Observable.invoke (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:637:20)
    at Results.Observable.trigger (eval at webpackJsonp.407.module.exports (scripts.bundle.js:1), <anonymous>:631:12)`